### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by RajeshReddyM**

### CODE REVIEW

**Comment by RajeshReddyM** [comment]
You are right, I have added a check to make sure the color doesn't repeat. Since I am generating each letter in the hexadecimal color using Math.random of a 16 digit letter I thought the probability of the color being repeated is very less nevertheless adding a check is always better.

File: src/colorChanger.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  // Gererates a Random Hexadecimal color code
+  var generateColor = function () {
+    var letters = '0123456789ABCDEF';
+    var color = '#';
+    for (var i = 0; i < 6; i++) {
+      color += letters[Math.floor(Math.random() * 16)];
+    }
+
+    if (colorArray.includes(color)) {
```
Comment (veem-abba) [inline src/colorChanger.js]: What is the time complexity of this? Is there a way we can speed it up?

File: src/colorChanger.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    var startTime = new Date().getTime();
+    // Changes the background color every second
+    var timer = setInterval(function() {
+      // Clears the timer andd exits after 10 seconds
+      if (new Date().getTime() - startTime > RUNTIME) {
+          clearInterval(timer);
+          return;
+      }
+      // change the background color of body
+      document.body.style.background = generateColor();
```
Comment (veem-abba) [inline src/colorChanger.js]: When I run this function in my console the color only changes 9 times instead of 10 times

File: src/colorChanger.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  // Gererates a Random Hexadecimal color code
+  var generateColor = function () {
+    var letters = '0123456789ABCDEF';
+    var color = '#';
+    for (var i = 0; i < 6; i++) {
+      color += letters[Math.floor(Math.random() * 16)];
+    }
+
+    if (colorArray.includes(color)) {
+      generateColor();
```
Comment (veem-abba) [inline src/colorChanger.js]: Is there another way we can do this without calling the function recursively? What is the downside of doing it this way?

File: src/colorChanger.js (hunk 1 above)
Comment (RajeshReddyM) [reply src/colorChanger.js]: The time complexity is O(n) as the Array.prototype.includes iterates over the n elements in the array to see if the element is already included. Using a hash map speeds it up as the time complexity is O(1) for a hash lookup.

File: src/colorChanger.js (hunk 3 above)
Comment (RajeshReddyM) [reply src/colorChanger.js]: I have refactored the method to not call this recursively. Calling a function recursively causes the stack overflow on a larger data set.

File: src/colorChanger.js (hunk 2 above)
Comment (RajeshReddyM) [reply src/colorChanger.js]: I have refactored the timer code and tested it using console it should work now.

**Pull request merged**